src/App.tsx

```diff
@@ -1,14 +1,52 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getAllTodos } from './api';
+import { StateContext } from './components/Store';
+import { TodosType } from './types/TodosType';
 
 export const App: React.FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [todosType, setTodosType] = useState(TodosType.all);
+  const { selectedTodo } = useContext(StateContext);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    if (todosType === TodosType.all) {
+      getAllTodos()
+        .then(setVisibleTodos)
+        .finally(() => setLoading(false));
+    } else if (todosType === TodosType.active) {
+      setVisibleTodos(allTodos.filter(todo => !todo.completed));
+    } else {
+      setVisibleTodos(allTodos.filter(todo => todo.completed));
+    }
+  }, [todosType, allTodos]);
```

Comment:
```suggestion
  const [todos, setTodos] = useState<Todo[]>([]);
  const [loading, setLoading] = useState(true);
  const [todosType, setTodosType] = useState(TodosType.all);
  const { selectedTodo } = useContext(StateContext);
  const [query, setQuery] = useState('');


const visibleTodos = useMemo(() => {
  switch(todosType) {
    case TodosType.active:
      return todos.filter(todo => !todo.completed)
     case TodosType.completed:
      return todos.filter(todo => todo.completed)
      default: 
        return todos;
  }
}, [todos, todosType])
```

Reply:
there is no sense to hold data in state if it can be calculated based on existing ones